From 24236c0eb5ddd710d1b28e63c6a7a685ef5660a3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20K=C3=B6ppeler?= <j.koeppeler@tu-berlin.de>
Date: Mon, 5 Jan 2026 17:29:02 +0100
Subject: [PATCH] tc: cake: add cake_mq support
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This adds support for the cake_mq variant of sch_cake to tc.

Signed-off-by: Jonas Köppeler <j.koeppeler@tu-berlin.de>
Signed-off-by: Toke Høiland-Jørgensen <toke@redhat.com>
Signed-off-by: David Ahern <dsahern@kernel.org>
---
 tc/q_cake.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/tc/q_cake.c
+++ b/tc/q_cake.c
@@ -667,6 +667,11 @@ static int cake_print_xstats(const struc
 			   " /%8u\n", GET_STAT_U32(MAX_ADJLEN));
 	}
 
+	if (st[TCA_CAKE_STATS_ACTIVE_QUEUES])
+		print_uint(PRINT_ANY, "active_queues",
+			   " active queues: %25u\n",
+			   GET_STAT_U32(ACTIVE_QUEUES));
+
 	if (st[TCA_CAKE_STATS_AVG_NETOFF])
 		print_uint(PRINT_ANY, "avg_hdr_offset",
 			   " average network hdr offset: %12u\n\n",
@@ -826,4 +831,11 @@ struct qdisc_util cake_qdisc_util = {
 	.parse_qopt	= cake_parse_opt,
 	.print_qopt	= cake_print_opt,
 	.print_xstats	= cake_print_xstats,
+};
+
+struct qdisc_util cake_mq_qdisc_util = {
+	.id		= "cake_mq",
+	.parse_qopt	= cake_parse_opt,
+	.print_qopt	= cake_print_opt,
+	.print_xstats	= cake_print_xstats,
 };
